lib/srv/desktop/windows_server.go

```diff
@@ -952,33 +919,40 @@ func (s *WindowsService) makeTDPSendHandler(ctx context.Context, emitter events.
 				// ones are around 2000 bytes. Anything approaching the limit of a single protobuf
 				// is likely some sort of DoS attempt and not legitimate RDP traffic, so we don't log it.
 				s.cfg.Log.Warnf("refusing to record %d byte PNG frame, image too large", len(b))
-			} else if err := emitter.EmitAuditEvent(ctx, e); err != nil {
-				s.cfg.Log.WithError(err).Warning("could not emit desktop recording event")
+			} else {
+				event, err := recorder.SetupSessionEvent(e)
+				if err == nil {
+					if err := recorder.RecordEvent(ctx, event); err != nil {
+						s.cfg.Log.WithError(err).Warning("could not record desktop recording event")
```

Comment:
nit: not sure if worth it to have a `recorder.SetupAndRecordEvent` as this repeat a few places.

Reply:
I think in places where the event will only be recorded this makes sense, done